olive/common/hf/model_io.py

```diff
@@ -124,6 +124,9 @@ def get_model_io_config(model_name: str, task: str, model: "PreTrainedModel", **
         for axis, axis_name in value.items():
             if axis_name == "past_sequence_length + 1":
                 value[axis] = "past_sequence_length + sequence_length"
+    # NOTE: Due to the complexity of dynamic_shapes, we don't provide it here.
```

Comment:
is the torch exporter able to combine dynamic axes and dynamic shapes? 

In the kv cache config, you add dynamic shapes for the kv cache, but we don't have dynamic shapes for the other inputs since it's not added here. Would that be a valid configuration?

Reply:
> is the torch exporter able to combine dynamic axes and dynamic shapes?
> 

Yes, kind of. We have a "do our best" mechanism to convert `dynamic_axes` to `dynamic_shapes` in dynamo exporter for torchscript users to have a quick way of "trying it out" feature that to sell it. But there is no guarantee that the conversion would 100% succeed though. Especially in `dynamic_shapes` we need users to provide specific min/max boundary of each dynamic dim.

> In the kv cache config, you add dynamic shapes for the kv cache, but we don't have dynamic shapes for the other inputs since it's not added here. Would that be a valid configuration?

Good catch on this. I am confused by the logics around `kv_cache` actually. I know there are two automation abouth `kv_cache`. I think one is from Optimum that Olive generates io_config from it if it's not provided, and HF model id is used. And the other one is that when `kv_cache` is set to True, Olive auto-gen `kv_cache` for users. 

My idea was that for the first approach, we don't generate `dynamic_shapes`, but only let dynamo exporter to trigger **from dynamic axes to dynamic shapes**, since Optimum does not have `dynamic_shapes` anyway. For the second one, in order to follow this promise to users (generate `kv_cache` for them), I wrote the conversion here. I didn't notice that could be conflict. Is it that `get_dynamic_shapes()` actually needs an initial version blocks it? If so, that should be fine? I only wanted to support non-HF auto cases. 

however, if HF cases are important, we can figure something out. I just didn't want to crash the export with broken `dynamic_shapes`, which is very likely to be the case, if we are going to generate it from Optimum config, as we need to auto-set up min and max of dynamic dims.

Reply:
the automatic kv cache addition applies to both user-provided io_config and hf model io_config from optimum. So, my concern with regards to the second case since we are missing dynamic shapes for the input_names coming from optimum created io_config. It only has dynamic shapes for the kv cache inputs. 

HF cases are most important for us since most models of interests are some huggingface causal lm models. Hopefully the conversion for dynamic axis to dynamic shape for the optimum produced io_config is simpler since there is no nesting structure for those.

Reply:
kv cache dynamic shape automation has been deleted. We have decided to go with the automation inside exporter. Inl Olive, we can just generate `dynamic_axes` and feed it to exporter.